src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionBase.cs

```diff
@@ -131,7 +131,7 @@ public string GetResponseHeaderValueWithCaching(HeaderDescriptor descriptor, Rea
             return
                 descriptor.Equals(KnownHeaders.Date) ? GetOrAddCachedValue(ref _lastDateHeaderValue, descriptor, value, valueEncoding) :
                 descriptor.Equals(KnownHeaders.Server) ? GetOrAddCachedValue(ref _lastServerHeaderValue, descriptor, value, valueEncoding) :
-                descriptor.GetHeaderValue(value, valueEncoding);
+                descriptor.GetHeaderValue(value, valueEncoding, replaceDangerousCharacters: true);
```

Comment:
I'm guessing you've added the `bool` to avoid the cost on HTTP/1.X?

`GetHeaderValue` is also being called directly from HTTP/2:
https://github.com/dotnet/runtime/blob/c4bc758c27c582223c51b9e6ba7ae4a14af4974a/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs#L711
https://github.com/dotnet/runtime/blob/c4bc758c27c582223c51b9e6ba7ae4a14af4974a/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs#L717

Reply:
> I'm guessing you've added the `bool` to avoid the cost on HTTP/1.X?

Actually no. It was a misunderstanding (I mistakenly attributed an intermittent test failure to the change).
I removed the parameter, since IMO it's better to replace `\0` in case of HTTP/1.X for consistency.

Assuming that the performance difference between `ContainsAny(SearchValues[3])` and `ContainsAny(byte)` does not lead to noticable difference E2E, I didn't special case HTTP/1.X.

Reply:
I wouldn't expect it to matter in E2E.
For reference, these numbers should still be a good approximation of how the different sets of values will impact searching throughput: [comment](https://github.com/dotnet/runtime/pull/106900#:~:text=Throughput%20numbers%20for%20scanning%20through%2010k%20elements%20(10k%20bytes%20or%2010k%20chars).%0ARows%20are%20ordered%20from%20fastest%20to%20slowest.)